### PR TITLE
Addition of Book Cipher Method

### PR DESCRIPTION
**Pull request by Viswalahiri**
Added a book cipher method in encoder.py

Added Source to README.md

Added 'book' choice to scram.py

### CODE REVIEW

File: encoder.py [hunk 1]
```diff
@@ -182,8 +196,13 @@ def calc(MSG, hash_type):
                 status += 1
 
         return ''.join(result_new)
+
```
Comment (ajitzero) [inline encoder.py]: Remove all these extra lines.

File: encoder.py [hunk 2]
```diff
@@ -48,7 +48,21 @@ def calc(MSG, hash_type):
 
         result = [str(bin(ord(elem))[2:]) for elem in MSG]
         return ' '.join(result)
-
+    
+     elif hash_type == 'book':
+        """Convert message to key representation"""
+        
+        passage = click.prompt("Enter passage", type=str)
+        w_l=passage.split(' ')
```
Comment (ajitzero) [inline encoder.py]: Follow spacing conventions as can seen in the rest of the code. You've used spaces correctly on Line 55 (above).
Spaces before and after for assignment, no spaces if written within a function call.

File: encoder.py [hunk 3]
```diff
@@ -48,7 +48,21 @@ def calc(MSG, hash_type):
...
+        key=[]
+        f_l_l=[a[0].lower() for a in w_l]
+        for i in MSG.lower():
+            if i not in f_l_l:
+                raise ValueError("Error: The entered passage does not support the message.")
+            if i in f_l_l:
+                key.append(str(f_l_l.index(i)+1))
+        return ','.join(key)
```
Comment (ajitzero) [inline encoder.py]: Correct me if I misunderstood something but shouldn't the result be separated by spaces instead of commas?

**Review by ajitzero — CHANGES_REQUESTED** [review]
Please make the appropriate changes, wherever applicable.

**Comment by ajitzero** [comment]
This should close #2 when merged.

File: encoder.py (hunk 1 above)
Comment (Viswalahiri) [reply encoder.py]: I've seen to it.

File: encoder.py (hunk 2 above)
Comment (Viswalahiri) [reply encoder.py]: encoder.py has been updated accordingly.

File: encoder.py (hunk 1 above)
Comment (ajitzero) [reply encoder.py]: Add the relevant commits, _then_ mark this as resolved.

File: encoder.py [hunk 4]
```diff
@@ -178,20 +182,16 @@ def calc(MSG, hash_type):
         domain = string.ascii_lowercase
         positions = [domain.find(elem) for elem in key]
         lenp = len(positions)
-
```
Comment (ajitzero) [inline encoder.py]: These four deletions have nothing to do with your additions, leave them be 🤦‍♂

File: encoder.py (hunk 3 above)
Comment (Viswalahiri) [reply encoder.py]: Both ways seem fine. Irrespective, the change has been implemented.

File: encoder.py (hunk 4 above)
Comment (Viswalahiri) [reply encoder.py]: Sorry about that. After I edited them, I noticed that the specific lines you asked for were from 182 - 196. I edited those lines but forgot to set the other ones I removed back.
 
Please don't mind.

**Review by ajitzero — APPROVED** [review]

**Comment by ajitzero** [comment]
@Viswalahiri Thanks for the contributions!

**Pull request merged**